source/developer/interactive-dialogs.rst

```diff
@@ -0,0 +1,403 @@
+
+  {
+    "display_name": "Assignee",
+    "name": "assignee",
+    "type": "select",
+    "data_source": "users"
+  }
+
+// XXX <add screenshot for an example>
+
+and for public channels, use
+
+  {
+    "display_name": "Post this message to",
+    "name": "channel",
+    "type": "select",
+    "data_source": "channels"
+  }
+
+// XXX <add screenshot for an example>
+
+The list of supported fields for the ``select`` type element is included below:
+
+.. csv-table::
+    :header: "Field", "Type", "Description"
+
+    "display_name", "String", "Display name of the field shown to the user in the dialog. Maximum 24 characters."
+    "name", "String", "Name of the field element used by the integration. Maximum 300 characters. You should use unique “name” fields in the same dialog."
+    "type", "String", "Set this value to ``select`` for a select element."
+    "data_source", "String", "(Optional) One of ``users``, or ``channels``. If none specified, assumes a manual list of options is provided by the integration."
+    "options", "String", "(Optional) An array of options for the select element. Not applicable for ``users`` or ``channels`` data sources."
+    "optional", "String", "(Optional) Set to true if this form element is not required. Default is ``false``."
+    "help_text", "Integer", "(Optional) Set help text for this form element. Maximum 150 characters."
+    "default", "Integer", "(Optional) Set a default value for this form element. Maximum 3,000 characters."
+    "placeholder", "Integer", "(Optional) A string displayed to help guide users in completing the element. Maximum 3,000 characters."
+
+Dialog Submission
+-----------------------
+
+When a user submits a dialog, Mattermost will perform client-side input validation to make sure
+
+  - All required fields are filled
+  - All formats are correct (e.g. email, telephone number, etc.)
+
+The submission payload sent to the integration is:
+
+.. code-block:: json
+
+  {
+      "type": "dialog_submission",
+      "callback_id": "<callback ID provided by the integration>",
+      "state": "<state provided by the integration>", 
+      "user_id": "<user ID of the user who submitted the dialog>",
+      "channel_id": "<channel ID the user was in when submitting the dialog>",
+      "team_id": "<team ID the user was on when submitting the dialog>",
+      "submission": {
+          "some_element_name": "<value of that element>",
+          "some_other_element": "<value of some other element>"
+      },
+      “cancelled”: false
+  }
+
+Optionally, the dialog can send an event back to the integration if ``notify_on_cancel`` parameter is set to true. If this happens, ``cancelled`` will be set to true on the above payload, and ``submission`` will be empty.
+
+Moreover, Mattermost also allows the integration itself to perform input validation. For example, you can require the user to enter a number between 0 and 10, and return the following response body if the condition isn't satisfied:
```

Comment:
```suggestion
Moreover, Mattermost also allows the integration itself to perform input validation. This can be done by responding to the dialog submission request with a JSON body containing an `errors` field. The `errors` field can contain a JSON object, mapping input field names to string error messages you would like to display to the user. For example, if you have a field named `num_between_0_and_10`, you can enforce the user to enter a number between 0 and 10 by returning the following response body if the condition isn't satisfied:
```